src/Generation/GapicClientGenerator.php

```diff
@@ -361,25 +362,34 @@ private function lroMethods(): Vector
 
     private function getClientDefaults(): PhpClassMember
     {
+        $clientDefaultValues = [
+        'serviceName' => AST::access(AST::SELF, $this->serviceName()),
+        // TODO: Understand if this should be named 'serviceAddress' or 'apiEndpoint'?
+        'serviceAddress' => AST::concat(AST::access(AST::SELF, $this->serviceAddress()), ':', AST::access(AST::SELF, $this->servicePort())),
+        'clientConfig' => AST::concat(AST::__DIR__, "/../resources/{$this->serviceDetails->clientConfigFilename}"),
+        'descriptorsConfigPath' => AST::concat(AST::__DIR__, "/../resources/{$this->serviceDetails->descriptorConfigFilename}"),
```

Comment:
For readability can we set these to local variables above?

```php
$clientConfigFilename = $this->serviceDetails->clientConfigFilename;
$descriptorConfigFilename = $this->serviceDetails->clientConfigFilename;

//...
    'clientConfig' => AST::concat(AST::__DIR__, "/../resources/{$clientConfigFilename}"),
    'descriptorsConfigPath' => AST::concat(AST::__DIR__, "/../resources/{$descriptorConfigFilename}"),
```

Reply:
Done.